src/foam/nanos/boot/Boot.java

```diff
@@ -58,25 +61,21 @@ public Boot(String datadir) {
 
     for ( int i = 0 ; i < l.size() ; i++ ) {
       NSpec sp = (NSpec) l.get(i);
+      XFactory factory = new SingletonFactory(new NSpecFactory((ProxyX) root_, sp));
```

Comment:
Why do we need to decorate in a SingletonFactory? Doesn't NSpecFactory already cache the service?

Reply:
Just add an invalidate() method on the NSpecFactory and call it instead.

Reply:
We are using SingletoneFactory when registering services adding invalidate() method on NSpecFactory would not help.

Reply:
Fixed.